src/sentry/models/organization.py

```diff
@@ -308,12 +336,30 @@ def default_owner_id(self):
         return self._default_owner_id
 
     def has_single_owner(self):
-        from sentry.models import OrganizationMember
+        owners = list(
+            self.get_members_with_org_roles([roles.get_top_dog().id]).values_list("id", flat=True)
+        )
+        return len(owners[:2]) == 1
+
+    def get_members_with_org_roles(self, roles: Collection[str]):
+        members_with_role = set(
+            self.member_set.filter(
+                role__in=roles,
+                user__isnull=False,
+                user__is_active=True,
+            ).values_list("id", flat=True)
```

Comment:
i could also not evaluate this immediately an instead do `union()` with two querysets for the return value